simulation/utils/generate_map.py

```diff
@@ -55,82 +57,101 @@ def get_seed(self):
         """Get the seed of the map"""
         return self.seed
 
-    def X(self):
+    def get_X_list(self):
         """Get list of x coordinates"""
-        xs = [point.x for point in self.points_before_interpolation] if not len(self.map_points) else [point[0] for point in self.map_points]
+        if not len(self.map_points):
+            xs = [point.x for point in self.points_before_interpolation]
+        else:
+            xs = [point[0] for point in self.map_points]
```

Comment:
```suggestion
            xs = self.points_before_interpolation[:, 0]
        else:
            xs = self.map_points[:, 0]
```
numpy daje nam możliwość dzielenia sobie macierzy szybciutko bez potrzeby iterowania, więc czemu by nie korzystać gdzie się da :)

Reply:
wiem, wiem, to jest jedna z rzeczy ktore zrobie, ale na razie tak jak pisalam sa same zmiany kosmetyczne w tym pull requeście :)